src/lcals-kokkos/PLANCKIAN-Kokkos.cpp

```diff
@@ -0,0 +1,69 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "PLANCKIAN.hpp"
+#if defined(RUN_KOKKOS)
+#include "common/KokkosViewUtils.hpp"
+#include <cmath>
+#include <iostream>
+
+namespace rajaperf {
+namespace lcals {
+
+void PLANCKIAN::runKokkosVariant(VariantID vid,
+                                 size_t RAJAPERF_UNUSED_ARG(tune_idx)) {
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  PLANCKIAN_DATA_SETUP;
+
+  auto x_view = getViewFromPointer(x, iend);
+  auto y_view = getViewFromPointer(y, iend);
+  auto u_view = getViewFromPointer(u, iend);
+  auto v_view = getViewFromPointer(v, iend);
+  auto w_view = getViewFromPointer(w, iend);
+
+  switch (vid) {
+
+  case Kokkos_Lambda: {
+
+    Kokkos::fence();
```

Comment:
Is the fence needed because memory movement is async?

Reply:
A `Kokkos::fence()` blocks on completion of all outstanding asynchronous Kokkos operations.
that includes parallel dispatch (e.g. parallel_for(), parallel_reduce()
and parallel_scan()).  Here we want to make sure that we do not time extraneous ascyn operations.